src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Validation.cs

```diff
@@ -153,10 +162,11 @@ internal static unsafe partial class Utf8Utility
 
                         continue; // need to perform a bounds check because we might be running out of data
 
-                    Sse2LoopTerminatedEarlyDueToNonAsciiData:
-
+                    LoopTerminatedEarlyDueToNonAsciiData:
+                        // x86 can only be little endian, while ARM can be big or little endian
+                        // so if we reached this label we need to check both combinations are supported
                         Debug.Assert(BitConverter.IsLittleEndian);
-                        Debug.Assert(Sse2.IsSupported);
+                        Debug.Assert(AdvSimd.Arm64.IsSupported || Sse2.IsSupported);
```

Comment:
>AdvSimd.Arm64.IsSupported  [](start = 37, length = 26)

Reading the comments, won't the assert be :
```csharp
Debug.Assert(AdvSimd.Arm64.IsSupported || (Sse2.IsSupported && BitConverter.IsLittleEndian));
```

Reply:
The line above it is checking for little endian. Is it ok if it stays like this?